src/components/KMultiselect/KMultiselect.vue

```diff
@@ -0,0 +1,911 @@
+        if (items.length) {
+          handleMultipleItemsSelect(items)
+        } else {
+          clearSelection()
+        }
+      }
+    })
+
+    watch(() => props.items, (newValue, oldValue) => {
+      // Only trigger the watcher if items actually change
+      if (JSON.stringify(newValue) === JSON.stringify(oldValue)) {
+        return
+      }
+
+      unfilteredItems.value = cloneDeep(props.items)
+      for (let i = 0; i < unfilteredItems.value.length; i++) {
+        // Ensure each item has a `selected` property
+        if (unfilteredItems.value[i].selected === undefined) {
+          unfilteredItems.value[i].selected = false
+        }
+
+        unfilteredItems.value[i].key = `${unfilteredItems.value[i].label?.replace(/ /gi, '-')?.replace(/[^a-z0-9-_]/gi, '')}-${i}` || `k-multiselect-item-label-${i}`
+        if (props.modelValue.includes(unfilteredItems.value[i].value) || unfilteredItems.value[i].selected) {
+          unfilteredItems.value[i].selected = true
+          selectedItems.value.push(unfilteredItems.value[i])
+          visibleSelectedItemsStaging.value.push(unfilteredItems.value[i])
+          unfilteredItems.value[i].key += '-selected'
+        }
+
+        stageSelections()
+      }
+
+      // Trigger an update to the popper element to cause the popover to redraw
+      // This prevents the popover from displaying "detached" from the KSelect
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      if (popper.value && typeof popper.value.updatePopper === 'function') {
+        nextTick(() => {
+          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+          // @ts-ignore
+          popper.value.updatePopper()
+        })
+      }
+    }, { deep: true, immediate: true })
+
+    return {
+      // keys and ids
+      key,
+      stagingKey,
+      multiselectId,
+      multiselectInputId,
+      multiselectTextId,
+      multiselectSelectedItemsStagingId,
+      multiselectSelectedItemsId,
+      // text
+      getPlaceholderText,
+      filterStr,
+      // items
+      unfilteredItems,
+      sortedItems,
+      selectedItems,
+      invisibleSelectedItemsStaging,
+      visibleSelectedItemsStaging,
+      invisibleSelectedItems,
+      visibleSelectedItems,
+      hiddenItemsTooltip,
+      // style and attributes
+      widthStyle,
+      inputWidth,
+      modifiedAttrs,
+      isHovered,
+      isFocused,
+      isDisabled,
+      isReadonly,
+      popper,
+      boundKPopAttributes,
+      // functions
+      sortItems,
+      handleItemSelect,
+      clearSelection,
+      triggerFocus,
+      onQueryChange,
+      onInputFocus,
+      onPopoverOpen,
+    }
+  },
+})
+</script>
+
+<style lang="scss" scoped>
+@import '@/styles/variables';
+@import '@/styles/functions';
+
+.k-multiselect {
+  width: fit-content; // necessary for correct placement of popup
+
```

Comment:
Let's add position: relative to keep the `.staging-area` confined.
```suggestion
  position: relative;

```